pennylane_lightning/core/src/utils/Memory.hpp

```diff
@@ -35,23 +36,28 @@ namespace Pennylane::Util {
  * @param bytes Number of bytes to allocate
  * @return Pointer to the allocated memory
  */
-inline auto alignedAlloc(uint32_t alignment, size_t bytes) -> void * {
+inline auto alignedAlloc(uint32_t alignment, size_t bytes,
+                         bool zero_init = false) -> void * {
     if (bytes % alignment != 0) {
         bytes = alignment * (bytes / alignment + 1);
     }
+    void *p = nullptr;
+
 #if defined(__clang__) && defined(__APPLE__)
     /*
      * We use `posix_memalign` for MacOS as Mac does not support
      * `std::aligned_alloc` properly yet (even in MacOS 10.15).
      */
-    void *p;
     posix_memalign(&p, alignment, bytes);
-    return p;
 #elif defined(_MSC_VER)
-    return _aligned_malloc(bytes, alignment);
+    p = _aligned_malloc(bytes, alignment);
 #else
-    return std::aligned_alloc(alignment, bytes);
+    p = std::aligned_alloc(alignment, bytes);
 #endif
+    if (zero_init) {
+        std::memset(p, 0, bytes);
+    }
+    return p;
```

Comment:
Nice addition, but I cannot find any actual use case of  `zero_init = True` in the codebase? Is there a followup PR?

Reply:
And this is why code-reviews are so important: The following chunk is missing:
```
        def _create_basis_state(self, index):
            """Return a computational basis state over all wires.
            Args:
                index (int): integer representing the computational basis state
            Returns:
                array[complex]: complex array of shape ``[2]*self.num_wires``
                representing the statevector of the basis state
            Note: This function does not support broadcasted inputs yet.
            """
            state = allocate_aligned_array(2**self.num_wires, np.dtype(self.C_DTYPE), True)
            state[index] = 1
            return self._reshape(state, [2] * self.num_wires)
```

I'll add that in.

Reply:
And done.